src/storage/txn/scheduler.rs

```diff
@@ -481,6 +546,46 @@ impl<E: Engine, L: LockMgr> MsgScheduler for Scheduler<E, L> {
             _ => unreachable!(),
         }
     }
+
+    fn on_batch_msg(&self, tasks: Vec<(u64, Msg)>) {
+        if tasks.is_empty() {
+            return;
+        }
+        let cid = match tasks[0].1 {
+            Msg::ReadFinished { cid, .. } => cid,
+            Msg::WriteFinished { cid, .. } => cid,
+            Msg::FinishedWithErr { cid, .. } => cid,
+            _ => panic!("msg type mismatch"),
+        };
+        self.on_batch_finish(
```

Comment:
Why use id of task[0]?  Should the id of  every task be same?

Reply:
yep, all sub-commands share the same command id.

Reply:
Do all commands in `process_batch_write_impl` share the same id?   But why do you still pass a vector of pair rather than `on_batch_msg(task_id: u64, results: Vec<Msg>)`?

Reply:
those are two different id. command id `cid` is assigned by scheduler, request id `id` is used by tidb to identify requests in a stream. I could work out a better name later.